### ISSUE TITLE
Working with Custom Elements and Events

### ISSUE THREAD

**Issue by alexlafroscia** [glimmerjs/glimmer.js#123]
I'm having a bit of trouble working with custom elements in an Ember app, and I _think_ it comes from some assumptions that Glimmer makes about angle-bracket components.

If I have some custom element (say `os-modal` for example) that I want to bind a custom event listener do, I want to do something like this:

```handlebars
<os-modal {{action 'onModalClose' on='modalclose'}}>
  Some modal content
</os-modal>
```

However, I get an error from Glimmer:

> Element modifiers are not allowed in components

The problem is that Glimmer has no way (that I know of) to tell the compiler that `os-modal` is not a Glimmer component, and to treat it like any other HTML element.

_Is_ there actually a way to tell Glimmer (or Ember) about non-Glimmer custom elements? If not, what would the steps be to work on that, and RFC for Glimmer? For Ember?

Original issue: https://github.com/ember-cli/ember-cli/issues/6925

**Comment by locks** [comment]
This is unrelated to custom elements, as the lack of element modifiers will make it not work on native HTML elements as well. If you check the [default project on Try](http://try.glimmerjs.com/), you will see that it has `<button onclick={{action increment}}>Click</button>` and not `<button {{action increment}}>Click</button>`

**Comment by alexlafroscia** [comment]
How do you bind a custom event then? I tried something like

```handlebars
<os-modal onmodalclose={{action “whatever”}}>
  Content
</os-modal>
```

But the event didn’t fire

**Comment by alexlafroscia** [comment]
[Here's an example of an Glimmer project that tries to establish a custom event listener the way that @locks recommended above](http://try.glimmerjs.com/?app=JTdCJTIyY29tcG9uZW50cyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJHbGltbWVyQXBwJTIyJTJDJTIydGVtcGxhdGUlMjIlM0ElN0IlMjJmaWxlTmFtZSUyMiUzQSUyMnNyYyUyRnVpJTJGY29tcG9uZW50cyUyRkdsaW1tZXJBcHAlMkZ0ZW1wbGF0ZS5oYnMlMjIlMkMlMjJzb3VyY2VUZXh0JTIyJTNBJTIyJTNDaDElM0VXZWxjb21lJTIwdG8lMjB0aGUlMjBHbGltbWVyJTIwUGxheWdyb3VuZCElM0MlMkZoMSUzRSU1Q24lM0NwJTNFWW91JTIwaGF2ZSUyMGNsaWNrZWQlMjB0aGUlMjBidXR0b24lMjAlN0IlN0Jjb3VudCU3RCU3RCUyMHRpbWVzLiUzQyUyRnAlM0UlNUNuJTNDZGl2JTIwb25teWN1c3RvbWV2ZW50JTNEJTdCJTdCYWN0aW9uJTIwaW5jcmVtZW50JTdEJTdEJTNFJTVDbiUyMCUyMCUzQ0N1c3RvbUV2ZW50QnV0dG9uJTNFJTVDbiUyMCUyMCUyMCUyMFNlbmQlMjBhJTIwY3VzdG9tJTIwZXZlbnQhJTVDbiUyMCUyMCUzQyUyRkN1c3RvbUV2ZW50QnV0dG9uJTNFJTVDbiUzQyUyRmRpdiUzRSUyMiU3RCUyQyUyMmNvbXBvbmVudCUyMiUzQSU3QiUyMmZpbGVOYW1lJTIyJTNBJTIyc3JjJTJGdWklMkZjb21wb25lbnRzJTJGR2xpbW1lckFwcCUyRmNvbXBvbmVudC50cyUyMiUyQyUyMnNvdXJjZVRleHQlMjIlM0ElMjJpbXBvcnQlMjBDb21wb25lbnQlMkMlMjAlN0IlMjB0cmFja2VkJTIwJTdEJTIwZnJvbSUyMCclNDBnbGltbWVyJTJGY29tcG9uZW50JyUzQiU1Q25leHBvcnQlMjBkZWZhdWx0JTIwY2xhc3MlMjBleHRlbmRzJTIwQ29tcG9uZW50JTIwJTdCJTVDbiUyMCUyMCU0MHRyYWNrZWQlMjBjb3VudCUyMCUzRCUyMDElM0IlNUNuJTIwJTIwaW5jcmVtZW50KCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwdGhpcy5jb3VudCUyQiUyQiUzQiU1Q24lMjAlMjAlN0QlNUNuJTdEJTIyJTdEJTdEJTJDJTdCJTIybmFtZSUyMiUzQSUyMkN1c3RvbUV2ZW50QnV0dG9uJTIyJTJDJTIydGVtcGxhdGUlMjIlM0ElN0IlMjJmaWxlTmFtZSUyMiUzQSUyMnNyYyUyRnVpJTJGY29tcG9uZW50cyUyRkN1c3RvbUV2ZW50QnV0dG9uJTJGdGVtcGxhdGUuaGJzJTIyJTJDJTIyc291cmNlVGV4dCUyMiUzQSUyMiUzQ2J1dHRvbiUyMG9uY2xpY2slM0QlN0IlN0JhY3Rpb24lMjBzZW5kQ3VzdG9tRXZlbnQlN0QlN0QlM0UlNUNuJTIwJTIwJTdCJTdCeWllbGQlN0QlN0QlNUNuJTNDJTJGYnV0dG9uJTNFJTIyJTdEJTJDJTIyY29tcG9uZW50JTIyJTNBJTdCJTIyZmlsZU5hbWUlMjIlM0ElMjJzcmMlMkZ1aSUyRmNvbXBvbmVudHMlMkZDdXN0b21FdmVudEJ1dHRvbiUyRmNvbXBvbmVudC50cyUyMiUyQyUyMnNvdXJjZVRleHQlMjIlM0ElMjJpbXBvcnQlMjBDb21wb25lbnQlMkMlMjAlN0IlMjB0cmFja2VkJTIwJTdEJTIwZnJvbSUyMCU1QyUyMiU0MGdsaW1tZXIlMkZjb21wb25lbnQlNUMlMjIlM0IlNUNuJTVDbmNvbnN0JTIwY3VzdG9tRXZlbnQlMjAlM0QlMjBuZXclMjBDdXN0b21FdmVudCgnbXlDdXN0b21FdmVudCclMkMlMjAlN0IlNUNuJTIwJTIwYnViYmxlcyUzQSUyMHRydWUlNUNuJTdEKSUzQiU1Q24lNUNuZXhwb3J0JTIwZGVmYXVsdCUyMGNsYXNzJTIwZXh0ZW5kcyUyMENvbXBvbmVudCUyMCU3QiU1Q24lMjAlMjBzZW5kQ3VzdG9tRXZlbnQoKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjB0aGlzLmVsZW1lbnQuZGlzcGF0Y2hFdmVudChjdXN0b21FdmVudCklM0IlNUNuJTIwJTIwJTdEJTVDbiU3RCUzQiUyMiU3RCU3RCU1RCUyQyUyMmhlbHBlcnMlMjIlM0ElNUIlNUQlN0Q=).

It doesn't seem like you can `onanycustomeventname` the way you can `onclick`, which doesn't provide you with a way to declaratively set up event listeners for custom events.

**Comment by jasonmit** [comment]
```hbs
<os-modal {{action 'onModalClose' on='modalclose'}}>
  Some modal content
</os-modal>
```

The newer versions of glimmer seem to no longer has this issue.  Within Ember, it was fixed in 3.1.0 most likely because of https://github.com/emberjs/ember.js/pull/15828

**Comment by wycats** [comment]
The latest version of the Glimmer VM uses capital letters rather than dasherized names as the signal that it's looking at a component.

This should fix this issue, I think.
